tests/Integration/Adapter/Tax/TaxComputerTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Integration\Adapter\Tax;
+
+use Country;
+use PrestaShop\Decimal\DecimalNumber;
+use PrestaShop\PrestaShop\Adapter\Tax\TaxComputer;
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
+use PrestaShop\PrestaShop\Core\Domain\TaxRulesGroup\ValueObject\TaxRulesGroupId;
+use State;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Tax;
+use TaxRule;
+use TaxRulesGroup;
+
+class TaxComputerTest extends KernelTestCase
+{
+    /**
+     * @var TaxComputer
+     */
+    private $taxComputer;
+
+    protected function setUp(): void
+    {
+        self::bootKernel();
+        $container = self::$kernel->getContainer();
+        $this->taxComputer = $container->get('prestashop.adapter.tax.tax_computer');
+    }
+
+    /**
+     * @dataProvider getComputePriceData
+     */
+    public function testComputePriceWithTaxes(float $taxRate, int $countryId, int $stateId, string $priceWithoutTaxes, string $priceWithTaxes): void
```

Comment:
Those new tests are really nice, thanks!
Now that the basic tests are here you should add some to test `getTaxRate`, especially for special use cases where we need to choose the default state